atomicfu/src/commonMain/kotlin/kotlinx/atomicfu/parking/ParkingDelegator.kt

```diff
@@ -0,0 +1,13 @@
+package kotlinx.atomicfu.parking
+
+/**
+ * Internal utility that delegates the thread suspending and resuming calls calls in the platform specific way (darwin, linux, windows).
+ * On jvm delegates to LockSupport.Park.
+ */
+internal interface ParkingDelegator {
```

Comment:
Unless I'm missing something, on each platform, there is a single implementation of this interface, so extracting it is premature, I feel.

```kotlin
internal expect class ParkingData

internal expect object ParkingDelegator {
    fun createRef(): ParkingData
    fun wait(ref: ParkingData) 
    fun timedWait(ref: ParkingData, nanos: Long)
    fun wake(ref: ParkingData)
    fun destroyRef(ref: ParkingData)
}
```
On the JVM, `internal actual typealias ParkingData = Unit` would achieve the same effect as what we have now.

Reply:
During development I had multiple delegator types, (futex, posix, spinning) and this allowed me to quickly switch between them in tests. But this can now indeed be removed.

Done.